packages/headless/src/features/commerce/recommendations/recommendations-actions.ts

```diff
@@ -0,0 +1,66 @@
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {AsyncThunkCommerceOptions} from '../../../api/commerce/commerce-api-client';
+import {CommerceRecommendationsRequest} from '../../../api/commerce/recommendations/recommendations-request';
+import {RecommendationsCommerceSuccessResponse} from '../../../api/commerce/recommendations/recommendations-response';
+import {isErrorResponse} from '../../../api/search/search-api-client';
+import {validatePayload} from '../../../utils/validate-payload';
+import {logQueryError} from '../../search/search-analytics-actions';
+import {
+  StateNeededByQueryRecommendationsCommerceAPI,
+  buildCommerceAPIRequest,
+} from '../common/actions';
+import {recommendationsSlotDefinition} from './recommendations';
+
+export interface QueryRecommendationsCommerceAPIThunkReturn {
+  /** The successful recommendations response. */
+  response: RecommendationsCommerceSuccessResponse;
+}
+
+const buildRecommendationCommerceAPIRequest = async (
+  slotId: string,
+  state: StateNeededByQueryRecommendationsCommerceAPI
+): Promise<CommerceRecommendationsRequest> => {
+  const commerceAPIRequest = await buildCommerceAPIRequest(state);
+  delete commerceAPIRequest.facets;
+  return {
+    ...commerceAPIRequest,
+    id: slotId,
+  };
+};
+
+export const fetchRecommendations = createAsyncThunk<
+  QueryRecommendationsCommerceAPIThunkReturn,
+  void,
+  AsyncThunkCommerceOptions<StateNeededByQueryRecommendationsCommerceAPI>
+>(
+  'commerce/recommendation/fetch',
+  async (_action, {getState, dispatch, rejectWithValue, extra}) => {
+    const state = getState();
+    const {apiClient} = extra;
+    const fetched = await apiClient.getRecommendations(
+      await buildRecommendationCommerceAPIRequest(
+        state.recommendations?.slotId || '',
```

Comment:
No need for an optional here, since we now have access to `recommendations` through `RecommendationsSection` in `StateNeededByQueryRecommendationsCommerceAPI`
```suggestion
        state.recommendations.slotId,
```